File: baseLib/src/androidTest/java/me/ycdev/android/lib/common/demo/service/operation/HelloOperation.java
package me.ycdev.android.lib.common.demo.service.operation;

import android.os.RemoteException;
import androidx.annotation.NonNull;

import java.util.concurrent.CountDownLatch;

import me.ycdev.android.lib.common.demo.service.IDemoService;
import me.ycdev.android.lib.common.ipc.IpcOperation;

public class HelloOperation implements IpcOperation<IDemoService> {
    private String mGift;
    private CountDownLatch mLatch;

    public HelloOperation(String gift) {
        mGift = gift;
    }

    public HelloOperation setNotifier(CountDownLatch latch) {
        mLatch = latch;
        return this;
    }

    @Override
    public void execute(@NonNull IDemoService service) throws RemoteException {
        service.sayHello(mGift);
        if (mLatch != null) {
            mLatch.countDown();
        }
    }

    @Override
    public String toString() {
        return "Operation[Hello]";
    }
}

File: baseLib/src/androidTest/java/me/ycdev/android/lib/common/demo/service/operation/HelloOperation.kt
package me.ycdev.android.lib.common.demo.service.operation

import android.os.RemoteException
import androidx.annotation.NonNull
import me.ycdev.android.lib.common.demo.service.IDemoService
import me.ycdev.android.lib.common.ipc.IpcOperation
import java.util.concurrent.CountDownLatch

class HelloOperation(private val mGift: String) : IpcOperation<IDemoService> {
    private var mLatch: CountDownLatch? = null

    fun setNotifier(latch: CountDownLatch): HelloOperation {
        mLatch = latch
        return this
    }

    @Throws(RemoteException::class)
    override fun execute(@NonNull service: IDemoService) {
        service.sayHello(mGift)
        if (mLatch != null) {
            mLatch!!.countDown()
        }
    }

    override fun toString(): String {
        return "Operation[Hello]"
    }
}
